### ISSUE TITLE
RichTextBox.Selection.GetPropertyValue(FontStyleProperty) returns DependencyProperty.UnsetValue for Hyperlink with Italic style

### ISSUE THREAD

**Issue by vonzshik** [dotnet/wpf#7178]
* .NET Core Version: .NET 6.0.9
* Windows version: Windows 10 21H2 (19044.2006)
* Does the bug reproduce also in WPF for .NET Framework 4.8?: Yes
 
 **Problem description:**
 
We use `RichTextBox.Selection.GetPropertyValue(FontStyleProperty)` to determine the selected text's style. The issue is, if the selected text is `Hyperlink` and it's fully selected, the returned style is incorrect if it's not default.

As an example for the repro below, the button's color depends on whether the selected text is `Italic` or not (blue if yes, otherwise red). By pressing the button the selected text's style is changed to/from `Italic`.

Here nothing is selected, the whole hyperlink's text is `Italic`.

![image](https://user-images.githubusercontent.com/14861420/194855110-99a1da9a-be61-4426-b8a0-b842d5eff241.png)

Here only a part of the hyperlink is selected, the button is blue.

![image](https://user-images.githubusercontent.com/14861420/194855243-7fcf2529-0459-498b-a743-d221f27f1096.png)

The whole hyperlink is selected, the button is red (even though the whole text is `Italic`).

![image](https://user-images.githubusercontent.com/14861420/194855389-1ad254f0-aeb9-4d5d-ab97-6bc54e509a49.png)

This happens because when the whole text is selected `GetPropertyValue(FontStyleProperty)` returns `DependencyProperty.UnsetValue`.

 **Actual behavior:**

`GetPropertyValue(FontStyleProperty)` returns `DependencyProperty.UnsetValue` for `Hyperlink` if has `Italic` font style.
 
 **Expected behavior:**

`GetPropertyValue(FontStyleProperty)` should return `FontStyles.Italic` for `Hyperlink` if has `Italic` font style.
 
 **Minimal repro:**

https://github.com/vonzshik/WpfRichTextBoxItalicLink

**Comment by lindexi** [comment]
@vonzshik Is it the Hyperlink style?

**Comment by miloush** [comment]
No, this happens as long as the starting position is included in the selection (i.e. even for the first character only). The reason is because the parent the property is read from is the Paragraph and the paragraph does not have italic set - basically the cursor is not inside a run, like this:

```xaml
<Paragraph>|<Hyperlink x:Name="HL"><Run Text="https://google.com"/></Hyperlink></Paragraph>
```

**Comment by miloush** [comment]
@singhashish-wpf we haven't really asked author any questions. I do think the handling could be improved on WPF side. One option is to ensure the positions are always as much inside as possible. This would, however, mean that if you select whole paragraph it would claim it is italic even if it only had italic runs but was not italic itself. That could be less than ideal for implementing something like format painter. Perhaps we could do the most top level parent that the start and end positions share, which is not a trivial fix though.

**Comment by miloush** [comment]
ping to keep this open @singhashish-wpf maybe there should be a label which would make the bot ignore issues that we think should be fixed but are on a backlog

**Comment by vonzshik** [comment]
> This submission has been automatically marked as stale because it has been marked as requiring author feedback but has not had any activity for 14 days.

I'm not sure what feedback is expected from me, but I would prefer for this issue to be fixed at some point :bee:
